### PR TITLE
Fix transction rollback on `Future` drop

### PR DESCRIPTION
**Pull request by wquark**
PR for https://github.com/sfackler/rust-postgres/issues/1120. Also rollsback the savepoint (found while implementing this).

### CODE REVIEW

File: tokio-postgres/src/client.rs [hunk 1]
```diff
@@ -480,8 +490,13 @@ impl Client {
                     return;
                 }
 
+                let query = if let Some(sp) = self.savepoint {
+                    format!("ROLLBACK TO {}", sp.name)
+                } else {
+                    "ROLLBACK".to_string()
+                };
```
Comment (sfackler) [inline tokio-postgres/src/client.rs:497]: nit: this'd be cleaner as a match IMO

**Comment by sfackler** [comment]
Looks good to me other than the one nit!

File: tokio-postgres/src/client.rs (hunk 1 above)
Comment (wquark) [reply tokio-postgres/src/client.rs:497]: I matched style with the other instances (e.g https://github.com/sfackler/rust-postgres/blob/64caf4c947cfde6c292f42406c046dce79f771bf/tokio-postgres/src/transaction.rs#L42-L46 and https://github.com/sfackler/rust-postgres/blob/64caf4c947cfde6c292f42406c046dce79f771bf/tokio-postgres/src/transaction.rs#L83-L87). Would you like to switch all of those or just this one?

Comment (sfackler) [reply tokio-postgres/src/client.rs:497]: Oh huh, let's keep it as-is then.

**Comment by sfackler** [comment]
Looks like it needs a `cargo fmt`, and this should fix the chrono deprecation warning:

```diff
diff --git a/postgres-types/src/chrono_04.rs b/postgres-types/src/chrono_04.rs
index d599bde0..64166309 100644
--- a/postgres-types/src/chrono_04.rs
+++ b/postgres-types/src/chrono_04.rs
@@ -111,9 +111,10 @@ impl ToSql for DateTime<FixedOffset> {
 impl<'a> FromSql<'a> for NaiveDate {
     fn from_sql(_: &Type, raw: &[u8]) -> Result<NaiveDate, Box<dyn Error + Sync + Send>> {
         let jd = types::date_from_sql(raw)?;
+        let jd = Duration::try_days(jd)?;
         base()
             .date()
-            .checked_add_signed(Duration::days(i64::from(jd)))
+            .checked_add_signed(jd)
             .ok_or_else(|| "value too large to decode".into())
     }
```

**Comment by wquark** [comment]
Ran `cargo fmt` and fixed warnings. Left the warning fixes in a separate commit since they seemed unrelated.

**Comment by wquark** [comment]
Hey @sfackler, would be great to get this pushed forward so we can remove our patched dependencies. Let me know if there are other blockers, thanks.

**Pull request closed without merging**